ExampleMod/Content/Tiles/Furniture/ExampleBed.cs

```diff
@@ -34,10 +40,27 @@ public class ExampleBed : ModTile
 			AddMapEntry(new Color(200, 200, 200), name);
 		}
 
-		public override bool HasSmartInteract() {
+		public override bool HasSmartInteract(int i, int j, SmartInteractScanSettings settings) {
 			return true;
 		}
 
+		public override void ModifySmartInteractCoords(ref int width, ref int height, ref int frameWidth, ref int frameHeight, ref int extraY) {
+			// Because beds have special smart interaction, this splits up the left and right side into the necessary 2x2 sections
+			width = 2; // Default to the Width defined for TileObjectData.newTile
+			height = 2; // Default to the Height defined for TileObjectData.newTile
+			extraY = 0; // Depends on how you set up frameHeight and CoordinateHeights and CoordinatePaddingFix.Y
+		}
+
+		public override void SetDrawPositions(int i, int j, ref int width, ref int offsetY, ref int height, ref short tileFrameX, ref short tileFrameY) {
+			height = 18; // Set frame height independent of CoordinateHeights
```

Comment:
This is already handled by CoordinateHeights, right? This doesn't seem to do anything.